site/docs/v1/tech/apis/identity-providers/links.adoc

```diff
@@ -0,0 +1,182 @@
+---
+layout: doc
+title: Link APIs
+description: The APIs for creating, retrieving, updating and deleting Identity Provider Links
+---
+
+== Overview
+
+[NOTE.since]
+====
+This API has been available since 1.28.0
+====
+
+This page contains the APIs that are used to manage Links that establish a relationship between a FusionAuth User and an Identity Provider.
+
+* <<Link a User>>
+* <<Complete a pending Link>>
+* <<Retrieve a Link>>
+* <<Unlink a User>>
+
+== Link a User
+
+This API is used to create a Link.
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Link a User
+[.endpoint]
+.URI
+--
+[method]#POST# [uri]#/api/identity-provider/link#
+--
+
+==== Request Body
+
+[.api]
+[field]#identityProviderId# [type]#[UUID]# [optional]#Optional#::
+The Id of the identity provider.
+
+[field]#identityProviderUserId# [type]#[String]# [optional]#Optional#::
+The Id or unique identifier for the User that is provided by the identity provider.
+
+[field]#userId# [type]#[UUID]# [required]#Required#::
+The FusionAuth Id of the User that is being linked to the identity provider.
+
+[source,json]
+.Example Request JSON
+----
+include::docs/src/json/identity-providers/links/create-link-request.json[]
+----
```

Comment:
I think this is showing the example usage of creating a user w/out a pending Link. If so, then I think all of these fields are required. If I only send in a `userId` - will this work?

Reply:
You are correct. These are required.